app-header/app-header.html

```diff
@@ -168,7 +168,14 @@
     :host([disabled])::after,
     :host([disabled]) #backgroundFrontLayer,
     :host([disabled]) #backgroundRearLayer,
-    :host([disabled]) ::content > app-toolbar:first-of-type {
+    :host([disabled]) ::content > app-toolbar:first-of-type,
+    :host([disabled]) ::content > [primary],
+    :host-context([disabled-scroll-effects]),
+    :host-context([disabled-scroll-effects])::after,
+    :host-context([disabled-scroll-effects]) #backgroundFrontLayer,
+    :host-context([disabled-scroll-effects]) #backgroundRearLayer,
+    :host-context([disabled-scroll-effects]) ::content > app-toolbar:first-of-type,
+    :host-context([disabled-scroll-effects]) ::content > [primary] {
```

Comment:
Since you were doing perf and measuring style recalc, it's probably worth to try switching this to `.disabled-scroll-effects` and see if it's better.